Description: Make compatible with wxPython 3.0
Author: Olly Betts <olly@survex.com>
Forwarded: no
Last-Update: 2014-08-18

--- a/apps/python/prompter/prompter.py
+++ b/apps/python/prompter/prompter.py
@@ -60,6 +60,8 @@
       self.config = config
       self.suggestions = suggestions
       wx.App.__init__(self, redirect)
+      # Suppress WXDEBUG assertions, as happens by default with wx2.8.
+      wx.App.SetAssertMode(self, wx.PYAPP_ASSERT_SUPPRESS)
 
    def OnInit(self):
       self.frame = PrompterFrame(parent=None, id=-1, title='pyprompter', version=self.version, config=self.config, suggestions=self.suggestions)
@@ -249,7 +251,7 @@
       # dialog is set up to change the current working directory to the path chosen.
       dlg = wx.FileDialog(
           self, message="Choose a file", defaultDir="", 
-          defaultFile="", wildcard=self.wildcard, style=wx.OPEN | wx.CHANGE_DIR
+          defaultFile="", wildcard=self.wildcard, style=wx.FD_OPEN | wx.FD_CHANGE_DIR
           )
       
       # Show the dialog and retrieve the user response. If it is the OK response, 
@@ -298,7 +300,7 @@
       # directory than the one initially set.
       dlg = wx.FileDialog(
           self, message="Save file as ...", defaultDir="", 
-          defaultFile="", wildcard=self.wildcard, style=wx.SAVE
+          defaultFile="", wildcard=self.wildcard, style=wx.FD_SAVE
           )
 
       # This sets the default filter that the user will initially see. Otherwise,
